docs/operational_guide/topology.md

```diff
@@ -0,0 +1,173 @@
+# Topology
+
+## Overview
+
+M3DB stores its topology (mapping of which hosts are responsible for which shards) in EtcD. There are three possible states that each host/shard pair can be in:
+
+1. Initializing
+2. Available
+3. Leaving
+
+Note that these states are not a reflection of the current status of an M3DB node, but an indicating of whether a given node has ever successfully bootstrapped and taken ownership of a given shard. For example, in a new cluster all the nodes will begin with all of their shards in the Initializing state. Once all the nodes finish bootstrapping, they will mark all of their shards as Available. If all the M3DB nodes are stopped at the same time, the cluster topology will still show all of the shards for all of the hosts as Available.
+
+## Initializing State
+
+The initializing state is the state in which all new host/shard combinations begin. For example, upon creating a new topology all the host/shard pairs will begin in the "Initializing" state and only once they have successfully bootstrapped will they transition to the "Available" state.
+
+The Initializing state is not limited to new topology, however, as it can also occur during topology changes. For example, during a node add/replace the new host will begin with all of its shards in the Initializing state until it can stream the data it is missing from its peers. During a node removal, all of the hosts who receive new shards (as a result of taking over the responsibilities of the node that is leaving) will begin with those shards marked as Initializing until they can stream in the data from the node leaving the cluster, or one of its peers.
+
+## Available State
+
+Once a node with a shard in the Initializing state successfully bootstraps all of the data for that shard, it will mark that shard as Available (for the single host) in the cluster topology.
+
+## Leaving State
+
+The leaving state indicates that a node is attempting to leave the cluster. The purpose of this state is to allow the node to remain in the cluster long enough for thhe nodes that are taking over its responsibilities to stream data from it.
+
+
+## Sample Cluster State Transitions - Node Add
+
+Replication factor: 3
+
+### Initial Topology
```

Comment:
Should the first state be all `I`? That's how we phrase it above

EDIT: Just realized this is for a node add. Maybe have another sample transition for a cluster init, but that might be pedantic

Reply:
Yeah I wanted to put that in, but it seemed like it might just be more confusing than anything because it starts with all I's and then transitions to all A's but its not really clear what happened or why it happened